force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -54,11 +65,18 @@ public with sharing class FlowControlLoopsCollections {
      * @param num2 The second number.
      * @return The maximum of the two numbers, or null if either number is null.
      */
+
     public static Integer findMax(Integer num1, Integer num2) {
+        if (num1 == num2) {
+            return num1; 
+        } else if (num1 == null || num2 == null) { 
+            return null; 
+        } else {
+            return Math.max(num1, num2);
+        }
         // Initialize largestNum as null
-        Integer largestNum = null;
-
-        return null; // Replace null with the variable you used to store the result
+        // Integer largestNum = Math.max();
+        // return largestNum; // Replace null with the variable you used to store the result
```

Comment:
good job. A good habit to get into is to remove any commented code before submitting a PR.